pysrc/pip_resolve.py

```diff
@@ -271,9 +271,10 @@ def matches_environment(requirement):
     sys_platform = sys.platform.lower()
     markers_text = get_markers_text(requirement)
     if markers_text and 'sys_platform' in markers_text:
-        match = SYSTEM_MARKER_REGEX.findall(markers_text)
-        if len(match) > 0:
-            return match[0].lower() == sys_platform
+        matches = SYSTEM_MARKER_REGEX.findall(markers_text)
+        lowercase_matches = [match.lower() for match in matches]
+        if len(lowercase_matches) > 0:
```

Comment:
In Python an empty list is considered Falsy and a list which has at least one element is considered Truthy
```suggestion
        if lowercase_matches:
```